src/main/java/org/ballerinax/datamapper/SampleDataAnalysisTask.java

```diff
@@ -71,260 +67,52 @@
 import java.util.stream.Stream;
 
 /**
- * Compiler extension to generate sample JSON files for data mapper.
+ * An {@code AnalysisTask} that is triggered for data mapper.
  */
-public class DataMapperPlugin extends AbstractCompilerPlugin {
-    public static final String URL_ENCODED_COLON = "%3A";
+public class SampleDataAnalysisTask implements AnalysisTask<CompilationAnalysisContext> {
```

Comment:
I think we can use SyntaxNodeAnalysisContext

Reply:
Will address in a future PR